drivers/flash/flash_stm32l5x.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2021 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_DOMAIN flash_stm32l5
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <soc.h>
+#include <stm32l5xx_ll_icache.h>
+#include <stm32_ll_system.h>
+
+#include "flash_stm32.h"
+
+#define STM32L5_SERIES_MAX_FLASH	512
+#define BANK2_OFFSET	(KB(STM32L5_SERIES_MAX_FLASH) / 2)
+
+void icache_clear_status(void)
+{
+	/* Clear any pending flags */
+	LL_ICACHE_WriteReg(FCR, LL_ICACHE_ReadReg(FCR) | ICACHE_FCR_CBSYENDF | ICACHE_FCR_CERRF);
+}
+
+/*
+ * offset and len must be aligned on 8 for write,
+ * positive and not beyond end of flash
+ */
+bool flash_stm32_valid_range(const struct device *dev, off_t offset,
+			     uint32_t len,
+			     bool write)
+{
+
+#if defined(FLASH_OPTR_DBANK) && (CONFIG_FLASH_SIZE < STM32L5_SERIES_MAX_FLASH)
+	/*
+	 * In case of bank1/2 discontinuity, the range should not
```

Comment:
Is that a possible case on L5 ?

Reply:
Some L552xx series has 256 KB and 512 KB flash size. All L562xx series has 512 KB flash size.